modules/ssh/ssh.go

```diff
@@ -217,6 +326,65 @@ func FetchContentsOfFileE(t *testing.T, host Host, useSudo bool, filePath string
 	return CheckSshCommandE(t, host, command)
 }
 
+func listFileInRemoteDir(t *testing.T, sshSession *SshSession, options ScpDownloadOptions) ([]string, error) {
+	logger.Logf(t, "Running command %s on %s@%s", sshSession.Options.Command, sshSession.Options.Username, sshSession.Options.Address)
+
+	var result []string
+
+	if err := setUpSSHClient(sshSession); err != nil {
+		return result, err
+	}
+
+	if err := setUpSSHSession(sshSession); err != nil {
+		return result, err
+	}
+
+	findCommand := fmt.Sprintf("find %s -type f", options.RemoteDir)
+
+	if options.FileNameFilter != "" {
+		findCommand = fmt.Sprintf("%s -name %s", findCommand, options.FileNameFilter)
+	}
+
+	if options.MaxFileSizeMB != 0 {
+		findCommand = fmt.Sprintf("%s -size -%dM", findCommand, options.MaxFileSizeMB)
+	}
+
+	r, err := sshSession.Session.Output(findCommand)
+
+	if err != nil {
+		return result, err
+	}
+	defer sshSession.Session.Close()
+
+	resultString := string(r)
+	resultString = resultString[:len(resultString)-1]
```

Comment:
Why?

Reply:
Unfortunately after doing the `find` command on the remote server, the resulting string of files always has a `\n` last character - so I'm cutting that off.

Added comment to code to explain:
```	
        // The last character returned is `\n` this results in an extra "" array
	// member when we do the split below. Cut off the last character to avoid
	// having to remove the blank entry in the array.
```